File: src/main/java/com/heroslender/viptools/DataBase/StorageCache.java
package com.heroslender.viptools.DataBase;

import com.heroslender.viptools.data.Loja;

import java.util.HashMap;
import java.util.Map;

public abstract class StorageCache implements Storage {
    // Cache
    private final Map<String, Loja> cache = new HashMap<>();

    @Override
    public Loja get(String dono) {
        return cache.get(dono);
    }
}

File: src/main/kotlin/com/heroslender/viptools/storage/StorageCache.kt
package com.heroslender.viptools.storage

import com.heroslender.viptools.loja.Loja
import org.bukkit.Location

open class StorageCache {
    val cache = mutableMapOf<String, Loja>()

    operator fun get(dono: String): Loja? {
        return cache[dono]
    }

    operator fun get(signLocation: Location): Loja? {
        return cache.values.firstOrNull { it.placas.contains(signLocation) }
    }
}